internal/types/types.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/satsuma-data/node-gateway/internal/config"
+	"go.uber.org/zap"
+)
+
+type UpstreamStatus struct {
+	BlockHeightCheck BlockHeightChecker
+	PeerCheck        Checker
+	SyncingCheck     Checker
+	ID               string
+	GroupID          string
+}
+
+// Provide the max block height found across node providers.
+func (s *UpstreamStatus) IsHealthy(maxBlockHeight uint64) bool {
+	if !s.PeerCheck.IsPassing() || !s.SyncingCheck.IsPassing() || !s.BlockHeightCheck.IsPassing(maxBlockHeight) {
+		zap.L().Debug("Upstream identifed as unhealthy.", zap.String("upstreamID", s.ID), zap.String("upstreamStatus", fmt.Sprintf("%+v", s)))
+
+		return false
+	}
+
+	zap.L().Debug("Upstream identifed as healthy.", zap.String("upstreamID", s.ID), zap.String("upstreamStatus", fmt.Sprintf("%+v", s)))
+
+	return true
+}
+
+//go:generate mockery --output ../mocks --name BlockHeightChecker --with-expecter
+type BlockHeightChecker interface {
```

Comment:
Good question - I meant to put a note about this in the PR description but never got to it. This was necessary to avoid a cyclic import error - I started using `UpstreamStatus` as a return value from the `HealthCheckManager`, and this resulted in the mock code depending on the `checks` package and causing cyclic imports.

Moving the types to a separate package is a pattern I've used in TypeScript for avoiding cyclic imports in the past, so I went with that. If you know of a more idiomatic Go way of doing that, I'd be happy to switch to it.

Reply:
Sorry thought I replied to this - yeah this is the pattern I've seen here as well.